2006Scape Server/src/main/java/com/rs2/GameEngine.java

```diff
@@ -135,7 +135,7 @@ public static void schedule(Tick tick) {
 	 * Port and Cycle rate.
 	 */
 	static {
-		serverlistenerPort = 43594;
+		serverlistenerPort = (GameConstants.SERVER_WORLD == 1) ? 43594 : 43597;
```

Comment:
_same as above_
> I'm not sure if this is the best way to handle it tbh, feels too hardcoded IMO.

Reply:
How should it be handled then? Cause it need to be identified as a separate client to the client for world 1 & any other way I thought of would make the bot client still able to connect to the non bot world

Reply:
Some sort of maths, so if the world is 3 it would still work without having to manually adjust all the values.

Reply:
Maybe even just something like this?
```suggestion
		serverlistenerPort = (GameConstants.SERVER_WORLD == 1) ? 43594 : 43595 + GameConstants.SERVER_WORLD;
```

Or is there anything stopping us going down in ports?
```suggestion
		serverlistenerPort = 43595 - GameConstants.SERVER_WORLD;
```

Reply:
Either one works but I prefer the idea of incrementing per world rather then subtracting as I feel it makes more sense, I'll commit the top change :)

Reply:
I agree it makes more sense to increment, but just feels odd jumping a few ports due to the fileserver